src/lib/Microsoft.Health.Events/Telemetry/Exceptions/EventHubConfigurationExceptionTelemetryProcessor.cs

```diff
@@ -0,0 +1,63 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using EnsureThat;
+using Microsoft.Health.Common.Telemetry;
+using Microsoft.Health.Logging.Telemetry;
+
+namespace Microsoft.Health.Events.Telemetry.Exceptions
+{
+    public class EventHubConfigurationExceptionTelemetryProcessor
+    {
+        private readonly HashSet<Type> _handledExceptions;
+
+        public EventHubConfigurationExceptionTelemetryProcessor()
+            : this(
+                typeof(InvalidEventHubException),
+                typeof(UnauthorizedAccessEventHubException))
+        {
+        }
+
+        public EventHubConfigurationExceptionTelemetryProcessor(params Type[] handledExceptionTypes)
+        {
+            _handledExceptions = new HashSet<Type>(handledExceptionTypes);
+        }
+
+        public bool HandleException(Exception ex, ITelemetryLogger logger, string connectorStage)
+        {
+            EnsureArg.IsNotNull(ex, nameof(ex));
+            EnsureArg.IsNotNull(logger, nameof(logger));
+
+            var exType = ex.GetType();
+
+            var lookupType = exType.IsGenericType ? exType.GetGenericTypeDefinition() : exType;
+
+            if (_handledExceptions.Contains(lookupType))
```

Comment:
Since we need to spend some more time thinking about handled exceptions and applying the `IHandle...` interface, I'm going to hold off applying your suggestion for now and I'll revisit it when we make these code changes.
(Also, since we don't expect subclasses of `InvalideEventHubException` and `UnauthorizedAccessEventHubExceptions`, since your comment I've marked these two classes with `sealed`.)